src/kafunk/Utility/AsyncSeq.fs

```diff
@@ -258,36 +258,45 @@ module AsyncSeq =
       [ s1 |> AsyncSeq.map Choice1Of4 ; s2 |> AsyncSeq.map Choice2Of4 ; s3 |> AsyncSeq.map Choice3Of4 ; s4 |> AsyncSeq.map Choice4Of4 ]
 
   /// merges multiple async seqs together, distribution spread across evenly throughout each async seq
-  let mergeAllRoundRobin (ss:AsyncSeq<'T> list) : AsyncSeq<'T> =
+  let mergeAllRoundRobin (asyncSeqList:AsyncSeq<'T> list) : AsyncSeq<'T> =
     asyncSeq { 
-      let n = ss.Length
-      if n > 0 then 
-        let ies = [| for source in ss -> source.GetEnumerator()  |]
-        let tasks = Array.zeroCreate n
-        for i in 0 .. ss.Length - 1 do 
-          let! task = Async.StartChildAsTask (ies.[i].MoveNext())
-          do tasks.[i] <- task
-        let fin = ref n
+      let numOfAsyncSeqs = asyncSeqList.Length
+      if numOfAsyncSeqs > 0 then 
+        let ies = [| for source in asyncSeqList -> source.GetEnumerator()  |]
+        
+        let! tasks =
+          asyncSeqList
+          |> List.toArray
+          |> Array.mapi(fun i _ -> async {
+            let! task = Async.StartChildAsTask (ies.[i].MoveNext())
```

Comment:
Wouldn't it be better to just loop over `ies` instead of `asyncSeqList` in this case? Saves you the extra lookup.